src/mpyl/steps/deploy/k8s/__init__.py

```diff
@@ -122,13 +122,13 @@ def substitute_namespaces(
     """
     env = env_vars.copy()
 
-    def get_namespace_for_linked_project(project_name: ProjectName):
+    def get_namespace_for_linked_project(project_name: ProjectName) -> Optional[str]:
         is_part_of_same_deploy_set = project_name in projects_to_deploy
         if is_part_of_same_deploy_set and pr_identifier:
             return f"pr-{pr_identifier}"
         return project_name.namespace
 
-    def replace_namespace(env_value, project_name, namespace):
+    def replace_namespace(env_value: str, project_name: str, namespace):
```

Comment:
```suggestion
    def replace_namespace(env_value: str, project_name: str, namespace: str) -> str:
```

Reply:
I want to understand why our setup is like that

Reply:
I think @Jorg88 centralized this logic? https://github.com/Vandebron/mpyl/blob/1b656ce198edf125eceeed4e94e0f69fd1a5c0f6/src/mpyl/steps/deploy/k8s/__init__.py#L26
Perhaps you could use that?

Reply:
good one!

Reply:
Looks good! You can still add the types btw